### DISCUSSION TITLE
Ray tracing in Next Week - Cornell Box

### DISCUSSION THREAD

**Discussion by trinhlehainam** [RayTracing/raytracing.github.io#852] [Debugging]
Hello everyone,

I'm so excited to learn ray tracing and did a lot of fun. But I'm stuck at rendering empty Cornell Box scene.

![image](https://user-images.githubusercontent.com/64761941/106610707-6cd9f080-65aa-11eb-9ae2-c996c06692d4.png)
![image (1)](https://user-images.githubusercontent.com/64761941/106613186-45385780-65ad-11eb-97e6-d8571384b996.png)

It has some darker color along the bright color and it's seem different with the image in the book.
Sorry if my English confuses you because I'm still learning it.

My source code about this project for someone want to take times to check.
https://github.com/trinhlehainam/StudyRayTracing

**Comment by trinhlehainam** [comment]
I solved this. 
The bright color is look like rectangle because the light that is reflected by Lambertian material is scattered randomly in unit cube (with center is the unit normal) . I fixed this by just allow ray is scattered in random point in unit sphere.
![image (2)](https://user-images.githubusercontent.com/64761941/106707236-3f398980-6634-11eb-875e-814983beb099.png)
![image (1)](https://user-images.githubusercontent.com/64761941/106707242-419be380-6634-11eb-8fbd-fbfb9c5cce98.png)

Final result
![image](https://user-images.githubusercontent.com/64761941/106707252-43fe3d80-6634-11eb-8a1a-2e3655321024.png)

**Reply by hollasch** [reply]
Ah! Congratulations on figuring this out.
